xds/internal/client/xds.go

```diff
@@ -170,9 +187,23 @@ func processHTTPFilterOverrides(cfgs map[string]*anypb.Any) (map[string]httpfilt
 	}
 	m := make(map[string]httpfilter.FilterConfig)
 	for name, cfg := range cfgs {
-		_, config, err := validateHTTPFilterConfig(cfg, false)
+		optional := false
+		if typeURL := cfg.GetTypeUrl(); typeURL == "type.googleapis.com/envoy.config.route.v3.FilterConfig" {
```

Comment:
Add a const for the string.

Reply:
Or, maybe use https://pkg.go.dev/github.com/golang/protobuf/ptypes#DynamicAny instead of manually check the type? We can do a type assertion on the message.

A similar change can be made for `unwrapHTTPFilterConfig()`, so you won't need the type string there.

Reply:
> Or, maybe use pkg.go.dev/github.com/golang/protobuf/ptypes#DynamicAny instead of manually check the type? We can do a type assertion on the message.

I just discovered `ptypes.Is`; used that instead.